ssl/statem/statem_lib.c

```diff
@@ -1007,6 +1007,28 @@ int ssl_choose_client_version(SSL *s, int version)
 }
 
 /*
+ * ssl_verify_client_session_version - Verify that provided SSL_SESSION matches
+ * the protocol negotiated with server.
+ *
+ * @s: client SSL handle.
+ *
+ * Returns 1 when it matches, 0 otherwise
+ */
+int ssl_verify_client_session_version(SSL *s)
+{
+  /* Empty session and no EAP-FAST */
+  if (s->session->session_id_length == 0 && !s->session->tlsext_tick) {
```

Comment:
Took some time to figure it out, but I think I got it. Put the `session->ssl_version = ` assignment back, but not sure which check you think should be removed.

I don't really want to go into more involved approach, but again it may be a good idea to do this after this PR. Right now, I'm trying to fix a bug that node.js users are experiencing.